### PR TITLE
feat: infinite scroll

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Move infinite scroll component from NNS-dapp.

# Changes

- lib: new component `<InfiniteScroll />` copied from NNS-dapp
- lib: fix `main` height to fit the all content
- docs: add page to documet components

### CODE REVIEW

File: src/docs/styles/docs-global.scss [hunk 1]
```diff
@@ -63,6 +64,10 @@ p code {
   font-size: inherit;
 }
 
+p strong {
+  color: #ffffff;
```
Comment (mstrasinskis) [inline src/docs/styles/docs-global.scss:68]: Will the docs support theming in the future?

**Review by mstrasinskis — APPROVED** [review]
LGTM 👍

**Comment by peterpeterparker** [comment]
@mstrasinskis by testing this PR I also noticed that the `main` element was not fitting well the height of its parent so I modified to `fit-content` and integrated it here.

File: src/docs/styles/docs-global.scss (hunk 1 above)
Comment (peterpeterparker) [reply src/docs/styles/docs-global.scss]: good point, I updated it with the contrasting color of the background:

```
p strong {
  color: var(--background-contrast);
}
```

**Pull request merged**